GOPKG_TOML_EXAMPLE.md

```diff
@@ -0,0 +1,60 @@
+# Gopkg.toml example
+
+## `[metadata]`
+ 1) Metadata defines metadata about the project that could be used by other independent systems. The metadata defined here will be ignored by dep.
+
+ 2) Metadata defines metadata about the dependency or override that could be used
+ by other independent systems. The metadata defined here will be ignored by dep.
```

Comment:
The two metadata sections in the existing example is for showing that `metadata` can exist at the root of manifest, as well as under `constraint` and `override`. Validation rules for `constraint` and `override` are the same.

So you can combine the above two points into a single paragraph and under `constraint` and `override` below, you can just add an example of metadata with them.